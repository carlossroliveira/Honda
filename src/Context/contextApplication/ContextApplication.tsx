/* eslint-disable @typescript-eslint/no-explicit-any */
// -------------------------------------------------
// Packages
// -------------------------------------------------
import React, { createContext, useContext, useState } from 'react';
import photo from '../../assets/GroupModal.svg';
import Swal from 'sweetalert2';
// -------------------------------------------------
// Hooks
// -------------------------------------------------
import { useFetch } from '../../hooks/useFetch';
// -------------------------------------------------
// Types
// -------------------------------------------------
import {
  IContextApplication,
  IContextApplicationProvider,
  IProductProps,
} from './types';

const Context = createContext({} as IContextApplication);

export const ThemeProviderApplication = ({
  children,
}: IContextApplicationProvider): JSX.Element => {
  const { data } = useFetch<IProductProps[]>('http://localhost:5000/info');

  const [array, setArray] = useState<any>([]);
  const [values, setValues] = useState<number>(0);

  if (values !== 4) {
    if (array.some((x: any) => array.indexOf(x) !== array.lastIndexOf(x))) {
      Swal.fire({
        text: `You've already added ${array.name} to your basket. Do you want to remove it ?`,
        showCancelButton: true,
        confirmButtonColor: '#21E7C5',
        confirmButtonText: 'Yes',
        cancelButtonText: 'No keep it',
        cancelButtonColor: '#C6C9D6',

        imageUrl: photo,
        imageWidth: 200,
        imageHeight: 200,
        imageAlt: 'Custom image',
      }).then(() => {
        array.pop();
      });
    }
  }

  const handleValue = () => {
    if (values < 4) {
      setValues(values + 1);
    } else {
      const Toast = Swal.mixin({
        toast: true,
        position: 'top-end',
        showConfirmButton: false,
        timer: 3000,
        timerProgressBar: true,
        didOpen: (toast) => {
          toast.addEventListener('mouseenter', Swal.stopTimer);
          toast.addEventListener('mouseleave', Swal.resumeTimer);
        },
      });

      Toast.fire({
        icon: 'error',
        title: 'Full cart!',
      });
    }
  };

  return (
    <Context.Provider value={{ data, handleValue, values, setArray, array }}>
      {children}
    </Context.Provider>
  );
};

export const useMyHookApplication = (): IContextApplication =>
  useContext(Context);
