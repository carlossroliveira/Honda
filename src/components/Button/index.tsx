// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ButtonSC } from './buttonStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { ButtonProps } from './types';

export const Button = (props: ButtonProps) => {
  return (
    <ButtonSC ClassName={false} onClick={props.onClick} role="button">
      <>
        {props.text}
        {props.icon}
      </>
    </ButtonSC>
  );
};
