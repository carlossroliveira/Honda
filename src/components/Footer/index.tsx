// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ParagraphSC } from './footerStyles';

export const Footer = () => {
  const currentYear: number = new Date().getFullYear();

  return <ParagraphSC>&copy; {currentYear} | Carlos Oliveira</ParagraphSC>;
};
