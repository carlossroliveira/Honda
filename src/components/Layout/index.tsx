// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------

// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './layoutStyles';

export const Layout = (): JSX.Element => {
  return (
    <ContainerSC>
      <h1>Layout</h1>
    </ContainerSC>
  );
};
