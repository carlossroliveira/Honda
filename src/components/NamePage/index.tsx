// -------------------------------------------------
// Packages
// -------------------------------------------------
import React, { useEffect } from 'react';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { INamePageProps } from './types';

export const NamePage = ({ title }: INamePageProps) => {
  useEffect(() => {
    document.title = `Reload Health - ${title}`;
  }, [title]);
  return <></>;
};
