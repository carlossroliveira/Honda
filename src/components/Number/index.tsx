// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { SPanSC } from './numberStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------
/* import { INumberProps } from './types'; */

export const Number = () => {
  return <SPanSC>{3}</SPanSC>;
};
