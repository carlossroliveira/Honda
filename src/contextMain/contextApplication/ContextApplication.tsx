// -------------------------------------------------
// Packages
// -------------------------------------------------
import React, { createContext, useContext, useState } from 'react';
// -------------------------------------------------
// Hooks
// -------------------------------------------------
import { useFetch } from '../../hooks/useFetch';
// -------------------------------------------------
// Types
// -------------------------------------------------
import {
  IContextApplication,
  IContextApplicationProvider,
  IProductProps,
} from './types';

const Context = createContext({} as IContextApplication);

export const ThemeProviderApplication = ({
  children,
}: IContextApplicationProvider): JSX.Element => {
  const { data } = useFetch<IProductProps[]>('http://localhost:5000/info');

  const [array, setArray] = useState<any>([]);
  const [values, setValues] = useState<number>(0);

  if (values !== 4) {
    if (array.some((x: any) => array.indexOf(x) !== array.lastIndexOf(x))) {
      console.log('a');
    }
  }

  const handleValue = () => {
    if (values < 4) {
      setValues(values + 1);
    }
  };

  return (
    <Context.Provider value={{ data, handleValue, values, setArray, array }}>
      {children}
    </Context.Provider>
  );
};

export const useMyHookApplication = (): IContextApplication =>
  useContext(Context);
