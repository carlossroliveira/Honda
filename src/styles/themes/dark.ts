export default {
  title: 'dark',

  fontFamily: {
    fontDefault: 'Roboto, sans-serif',
  },
  color: {
    primary: '#ff8d66',
    secondary: '#393E46',
    tertiary: '',
    quaternary: '#F5F7FE',
  },
  background: {
    primary: '#F5F7FE',
    secondary: '#393E46',
    tertiary: '',
    quaternary: '#F5F7FE',
  },
  boxShadow: {
    primary:
      'rgba(50, 50, 93, 0.25) 0px 50px 100px -20px, rgba(0, 0, 0, 0.3) 0px 30px 60px -30px, rgba(10, 37, 64, 0.35) 0px -2px 6px 0px inset',
    secondary: '0px 2px 2px rgba(84, 89, 118, 0.46)',
  },
};
