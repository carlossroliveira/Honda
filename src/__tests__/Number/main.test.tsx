// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import { ThemeProvider } from 'styled-components';
import { render, screen } from '@testing-library/react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { Number } from '../../components/Number';
// -------------------------------------------------
// Context
// -------------------------------------------------
import { ThemeProviderApplication } from '../../context/contextApplication/ContextApplication';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import dark from '../../styles/themes/dark';

describe('Button Testing', () => {
  it('Should render the component correctly.', () => {
    render(
      <ThemeProvider theme={dark}>
        <ThemeProviderApplication>
          <Number score={0} />
        </ThemeProviderApplication>
      </ThemeProvider>,
    );

    expect(screen.getByText(/0/i)).toBeInTheDocument();
  });

  it('Should render with the value 10 correctly.', () => {
    render(
      <ThemeProvider theme={dark}>
        <ThemeProviderApplication>
          <Number score={10} />
        </ThemeProviderApplication>
      </ThemeProvider>,
    );

    expect(screen.getByText(/10/i)).toBeInTheDocument();
  });

  it('component rendering matches snapshot correctly', () => {
    render(
      <ThemeProvider theme={dark}>
        <ThemeProviderApplication>
          <Number score={10} />
        </ThemeProviderApplication>
      </ThemeProvider>,
    );
    expect(
      render(
        <ThemeProvider theme={dark}>
          <ThemeProviderApplication>
            <Number score={10} />
          </ThemeProviderApplication>
        </ThemeProvider>,
      ),
    ).toMatchSnapshot();
  });
});
