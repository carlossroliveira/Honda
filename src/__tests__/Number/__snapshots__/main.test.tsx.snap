// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Number Testing component rendering matches snapshot correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  background: linear-gradient(180.32deg,#810000 15.34%,#ff8d66 98.15%);
  box-shadow: 0px 2px 2px rgba(84,89,118,0.46);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 26px;
  height: 26px;
  border-radius: 50%;
  font-size: 14px;
  font-weight: 400;
  line-height: 39px;
  text-transform: capitalize;
  color: #F5F7FE;
}

@media (max-width:768px) {
  .c0 {
    background: #cc0000;
  }
}

<body>
    <div>
      <span
        class="c0"
      >
        10
      </span>
    </div>
    <div>
      <span
        class="c0"
      >
        10
      </span>
    </div>
  </body>,
  "container": .c0 {
  background: linear-gradient(180.32deg,#810000 15.34%,#ff8d66 98.15%);
  box-shadow: 0px 2px 2px rgba(84,89,118,0.46);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 26px;
  height: 26px;
  border-radius: 50%;
  font-size: 14px;
  font-weight: 400;
  line-height: 39px;
  text-transform: capitalize;
  color: #F5F7FE;
}

@media (max-width:768px) {
  .c0 {
    background: #cc0000;
  }
}

<div>
    <span
      class="c0"
    >
      10
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
